tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/EquinoxResolver.java

```diff
@@ -81,31 +81,47 @@
 @Component(role = EquinoxResolver.class)
 public class EquinoxResolver {
 
-    //The following properties are not supported for general usage and intended to experimental testing when developing Tycho, so use with care
+    private static final String FORCE_KEEP_USES = "First attempt at resolving bundle failed. Trying harder by keeping `uses` information... This may drastically slow down your build!";
 
-    /**
-     * If set to true this keep the 'uses' constraints of a package, this will make it more hard for
-     * the resolver or even let him fail to compute a solution if different package providers are
-     * present
-     */
-    private static final boolean KEEP_USES = Boolean.getBoolean("tycho.equinox.resolver.uses");
-    /**
-     * Keep the default batch size, but allow to override this if necessary, if 'uses' constrains
-     * are kept, do not restrict the batch size as this potentially fails the resolve
-     */
-    private static final String BATCH_SIZE = System.getProperty("tycho.equinox.resolver.batch.size",
-            KEEP_USES ? null : "1");
-    /**
-     * Set the batch timeout to an acceptable timeout before fallback to resolve one bundle at a
-     * time, but allow to override this if necessary
-     */
-    private static final String BATCH_TIMEOUT = System.getProperty("tycho.equinox.resolver.batch.timeout",
-            String.valueOf(TimeUnit.SECONDS.toMillis(30)));
+    protected static final class ResolverConfiguration {
+        //The following properties are not supported for general usage and intended to experimental testing when developing Tycho, so use with care
 
-    /**
-     * Allow to adjust the default thread count used in resolver operations
-     */
-    private static final int THREAD_COUNT = Integer.getInteger("tycho.equinox.resolver.executor.threads", 1);
+        public ResolverConfiguration() {
+            // keep default values
+        }
+
+        public void forceKeepUses() {
+            if (!keepUses) {
+                keepUses = true;
+                batchSize = System.getProperty("tycho.equinox.resolver.batch.size", keepUses ? null : "1");
```

Comment:
I don't think we save anything when conditionally lookup the system property here, whats wrong with the static final ones?